util/client.go

```diff
@@ -61,6 +61,11 @@ func (hc *HeaderClient) Do(req *http.Request) (*http.Response, error) {
 			req.Header.Add(key, v)
 		}
 	}
+
+	// Use default user agent if none is set
+	if userAgent := hc.Header.Get("User-Agent"); userAgent == "" {
+		req.Header.Add("User-Agent", "csaf-distribution "+SemVersion)
```

Comment:
The generated user agent must follow https://datatracker.ietf.org/doc/html/rfc9110#name-user-agent which it currently does not.